src/components/buttons/icon-button/icon-button.tsx

```diff
@@ -32,7 +52,20 @@ const getIconThemeColor = props => {
   return props.icon.props.theme;
 };
 
-export const IconButton = props => {
+export const IconButton: React.FC<IconButtonProps> = props => {
+  invariant(
+    props.isToggleButton && typeof props.isToggled === 'undefined',
+    'ui-kit/IconButton: when `isToggleButton` is passed `isToggled` is required.'
+  );
+  invariant(
+    !props.isToggleButton && typeof props.isToggled !== 'undefined',
+    'ui-kit/IconButton: when not `isToggleButton` `isToggled` does not affect the component.'
+  );
+  invariant(
+    !props.isToggleButton && props.theme !== 'default',
+    'ui-kit/IconButton: when not `isToggleButton` a `theme` is not supported.'
+  );
```

Comment:
I kind of even like this more. Discriminated union types won't work here as one type depends on another and it's not just an or.

Reply:
That's nicer yes!